scripts/simulate_live_data.py

```diff
@@ -24,13 +25,32 @@ def copy_dataset(source_dir, dest_dir, rate):
         source_item = os.path.join(source_dir, item)
         dest_item = os.path.join(dest_dir, item)
         if os.path.isfile(source_item):
-            shutil.copy(source_item, dest_item)
+            if not slow_copy_mode:
+                shutil.copy(source_item, dest_item)
+            else:
+                slow_copy(source_item, dest_item)
 
             time.sleep(float(rate))
         if os.path.isdir(source_item):
             copy_dataset(source_item, dest_item, rate)
 
 
+def slow_copy(source, dest):
+    with open(source, "rb") as source_file:
+        data = source_file.read()
+    data_size = len(data)
+
+    if os.path.exists(dest):
+        os.remove(dest)
+
+    with open(dest, "wb") as dest_file:
+        writen = 0
+        while writen < data_size - 1000:
+            start = writen
+            stop = start + data_size // 10
+            writen += dest_file.write(data[start:stop])
```

Comment:
If I understand this correctly (I may not have), to calculate the start and stop, Instead of `data_size // 10`, should it be:
```Python
with open(dest, "wb") as dest_file:
        written = 0
        chunk_size = data_size // 10
        while written < data_size:
            start = written
            stop = written + chunk_size
            dest_file.write(data[start:stop])
            written += chunk_size
```
to determine the stop position for each chunk?

Reply:
Oops, that ` - 1000` should not have been committed. I was trying to force some errors by skipping the last block.

That should do roughly the same. Though the last chunk will probably be a different size from the others (if the file size is not a multiple of 10), so the file value of `written` would be wrong. This would not matter much is would be larger than data_size and would stop the loop.

`write()` returns the number of bytes written, so the value should be correct. I'll admit it is a bit C-ish, but I think its more correct.